### PR TITLE
Added "fun" to various Kotlin interfaces and updates SDKs a bit

### LINKED ISSUE #102: [BUG] - setOnSetImageUriCompleteListener can't use lambda form anymore for the interface OnSetImageUriCompleteListener
- Lib Version [e.g. 1.1.0]
 2.3.0

**Describe the bug**
When it was in Java, I could use :

```
cropImageView.setOnSetImageUriCompleteListener { _, _, error ->
...
```

Now I have to use :

```kt
cropImageView.setOnSetImageUriCompleteListener(object : CropImageView.OnSetImageUriCompleteListener {
    override fun onSetImageUriComplete(view: CropImageView, uri: Uri, error: Exception?) {

```

**To Reproduce**
1. Just try the code I wrote above, of before this version and after.

**Expected behavior**
Should still allow to do it.

**Screenshots**
No need

**Smartphone (please complete the following information):**
None.

**Additional context**
I think whoever converted to Kotlin didn't look enough at the resulting code...
To fix this, I think it's possible to use "fun interface" instead of just "interface". This should be added for each interface in the project that has a single function.

### PR DESCRIPTION
**Pull request by AndroidDeveloperLB**
close #102 

### Cause:
When refactoring from java we lost the ability of using lambda on our callbacks 

### How the bug was solved:
Add `fun` notation to the interfaces.

### CODE REVIEW

**Review by Canato — COMMENTED** [review]
Nice work! Thanks

Can you please add the ticket: 
`https://github.com/CanHub/Android-Image-Cropper/issues/102`
To the Changelog file? 

Btw, the CI is broken =/ 

On a side not, should we change the sample code so we are sure this doesn't break again?

File: sample/src/main/java/com/canhub/cropper/sample/options_dialog/app/SOptionsDialogBottomSheet.kt [hunk 1]
```diff
@@ -189,8 +188,10 @@ internal class SOptionsDialogBottomSheet : BottomSheetDialogFragment(), SOptions
         when (options.cropShape) {
             CropImageView.CropShape.RECTANGLE -> binding.cropShape.chipRectangle.isChecked = true
             CropImageView.CropShape.OVAL -> binding.cropShape.chipOval.isChecked = true
-            CropImageView.CropShape.RECTANGLE_VERTICAL_ONLY -> binding.cropShape.chipRectangleVerticalOnly.isChecked = true
-            CropImageView.CropShape.RECTANGLE_HORIZONTAL_ONLY -> binding.cropShape.chipRectangleHorizontalOnly.isChecked = true
+            CropImageView.CropShape.RECTANGLE_VERTICAL_ONLY -> binding.cropShape.chipRectangleVerticalOnly.isChecked =
+                true
+            CropImageView.CropShape.RECTANGLE_HORIZONTAL_ONLY -> binding.cropShape.chipRectangleHorizontalOnly.isChecked =
+                true
```
Comment (Canato) [inline sample/src/main/java/com/canhub/cropper/sample/options_dialog/app/SOptionsDialogBottomSheet.kt:194]: This changes broke the ktlintCI

Comment (AndroidDeveloperLB) [reply sample/src/main/java/com/canhub/cropper/sample/options_dialog/app/SOptionsDialogBottomSheet.kt:194]: How so? Why did it break anything? What is the change here, anyway? I don't see any error here on the IDE... The sample works fine. Everything builds fine...

Comment (AndroidDeveloperLB) [reply sample/src/main/java/com/canhub/cropper/sample/options_dialog/app/SOptionsDialogBottomSheet.kt]: Anyway, I've made it have a new line as the logs that you've shown say. Still a very weird issue.

**Comment by AndroidDeveloperLB** [comment]
I didn't create anything in the changelog file. Should I ?

File: sample/src/main/java/com/canhub/cropper/sample/options_dialog/app/SOptionsDialogBottomSheet.kt (hunk 1 above)
Comment (Canato) [reply sample/src/main/java/com/canhub/cropper/sample/options_dialog/app/SOptionsDialogBottomSheet.kt]: The error message:

```
> Task :sample:ktlintMainSourceSetCheck FAILED
/home/runner/work/Android-Image-Cropper/Android-Image-Cropper/sample/src/main/java/com/canhub/cropper/sample/options_dialog/app/SOptionsDialogBottomSheet.kt:191:62 Missing newline after "->"
/home/runner/work/Android-Image-Cropper/Android-Image-Cropper/sample/src/main/java/com/canhub/cropper/sample/options_dialog/app/SOptionsDialogBottomSheet.kt:193:64 Missing newline after "->"
```

**The Ci link:** https://github.com/CanHub/Android-Image-Cropper/pull/104/checks?check_run_id=2239558660


Screenshot 2021-04-01 at 09 51 59

**Comment by Canato** [comment]
> I didn't create anything in the changelog file. Should I ?

Yes please, More into the [CONTRIBUTING](https://github.com/CanHub/Android-Image-Cropper/blob/main/CONTRIBUTING.md) file

File: sample/src/main/java/com/canhub/cropper/sample/options_dialog/app/SOptionsDialogBottomSheet.kt (hunk 1 above)
Comment (AndroidDeveloperLB) [reply sample/src/main/java/com/canhub/cropper/sample/options_dialog/app/SOptionsDialogBottomSheet.kt]: It shows it even though I've added a new line now?

Comment (Canato) [reply sample/src/main/java/com/canhub/cropper/sample/options_dialog/app/SOptionsDialogBottomSheet.kt]: Now is ok, you can see on the bottom of the github PR page =D

**Comment by AndroidDeveloperLB** [comment]
OK I've updated it. Hope it's how it's supposed to be.

File: sample/src/main/java/com/canhub/cropper/sample/options_dialog/app/SOptionsDialogBottomSheet.kt (hunk 1 above)
Comment (AndroidDeveloperLB) [reply sample/src/main/java/com/canhub/cropper/sample/options_dialog/app/SOptionsDialogBottomSheet.kt]: How did you perform this check? How come this issue even exists, while the IDE is completely fine with it?

**Review by Canato — APPROVED** [review]

**Comment by AndroidDeveloperLB** [comment]
Now what will happen?

**Comment by AndroidDeveloperLB** [comment]
I'm so confused. What's going on?

**Comment by Canato** [comment]
> I'm so confused. What's going on?

I approved, was waiting CI to run, have a meeting in my work, merge it, create the release, JitPack build the package and make it available. 

I believe should be ready for usage =D

**Comment by AndroidDeveloperLB** [comment]
I had a lot of emails. What is "CI", and how did you get the error before, while I didn't have any error on the IDE?

**Comment by Canato** [comment]
> I had a lot of emails. What is "CI", and how did you get the error before, while I didn't have any error on the IDE?

CI means continuous integration. Is a serie of automated steps that keep our codeStyle. We use ktlint.

Normally when projects use CodeStyle you can see it on the `.idea/CodeStyle` for that your Android Studio Preferences need to be set into the project CodeStyle, then you can run the commands of `./gradlew ktlint` 

Another way to enforce this without commands, and the way most people use, select the whole file `cmd+a` and click `^+cmd+i` and `option+cmd+L`, not sure in other OS's, so the code will format. 

Later the CI will break to keep the code style in place, some times most people just throw in the CI and wait to see if fails, depends on the Dev

**Comment by AndroidDeveloperLB** [comment]
So somehow the code style of the project was imported to the IDE, but the IDE failed to notice it, and didn't show a warning about it or format it as it wanted?
I thought it's all automatic anyway, that when I open a project, it uses its code style that the original developer made for it, no?

**Comment by Canato** [comment]
> So somehow the code style of the project was imported to the IDE, but the IDE failed to notice it, and didn't show a warning about it or format it as it wanted?
> I thought it's all automatic anyway, that when I open a project, it uses its code style that the original developer made for it, no?

It won't show any warning, sadly that's now hot it works =/. 

You need to run the commands, or the shorcuts or the terminal ones.

**Comment by AndroidDeveloperLB** [comment]
But shouldn't it auto-format anyway to the rules that were set?

**Comment by Canato** [comment]
> But shouldn't it auto-format anyway to the rules that were set?

It should when using the commands, if is not, maybe the AS is getting the wrong Style file

**Comment by AndroidDeveloperLB** [comment]
ok

**Pull request merged**